File: Python/矩阵转置.py
import sys
import math
x,k,a=int(math.sqrt(len(sys.argv))),1,[]
for i in range(x):
	temp=[]
	for j in range(x):
		temp.append(int(sys.argv[k]))
		k+=1
	a.append(temp)
with open('result.txt','w') as fp:
	for i in range(x):
		for j in range(x):
			fp.write(str(a[j][i])+' ')

File: C/矩阵转置.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>
int main(int argc,char** argv)
{
	FILE *fp;
	if((fp=fopen("result.txt","w"))==NULL)
		return -1;
	int x,i,j,k=1;
	x=sqrt(argc);
	int a[x][x];
	for(i=0;i<x;i++)
	{
		for(j=0;j<x;j++)
		{
			a[i][j]=atoi(argv[k]);
			k++;
		}
	}
	for(i=0;i<x;i++)
	{
		for(j=0;j<x;j++)
		{
			fprintf(fp,"%d ",a[j][i]);
		}
	}
	fclose(fp);	
	return 0;
}
